### ISSUE TITLE
Using Ember.set on an Ember.Object causes Object.keys to report toString as a key.

### ISSUE THREAD

**Issue by martndemus** [emberjs/ember.js#13050]
I'm seeing this happen in 2.5.0-beta.1
See this [twiddle](https://ember-twiddle.com/955aa87bb0ec4830cb9b)

```js
let object = Ember.Object.create();
Object.keys(object) // => [];
Ember.set(object, 'one', 'foo');
Object.keys(object) // => ['toString', 'one'];
```

This is particularly annoying when using the `{{each-in}}` helper.

**Comment by stefanpenner** [comment]
Woah

**Comment by martndemus** [comment]
@stefanpenner fix pls, yes?

**Comment by rwjblue** [comment]
Does the same happen with this.set? Other POJO style objects? Or only Ember.Object and its descendants?

**Comment by stefanpenner** [comment]
likely this https://github.com/emberjs/ember.js/commit/f9b99cdc0cacbeb6d6105cdc9ff0af9395bcf873

**Comment by martndemus** [comment]
@rwjblue Yes `object.set` also makes this happen.

**Comment by martndemus** [comment]
Could this be the offender? [fa875177](https://github.com/emberjs/ember.js/commit/fa8751770ff707f38770a36193c761f5226b41d2)

**Comment by stefanpenner** [comment]
https://github.com/emberjs/ember.js/commit/3a2486eb911ad77d2ab1a4324c1aacef08c156da so this commit was revealed in bisect

**Comment by bcardarella** [comment]
DockYard giveth, DockYard taketh away

**Comment by duggiefresh** [comment]
> Which is largely because toString memoizes itself (which has been around for ages). I suspect though, we may want to rethink this pattern...

Hola! What do you have in mind, Stefan? I have some bandwidth and can take a crack at this. Thanks!

**Comment by stefanpenner** [comment]
I would like to see some experiments that remove the memoization and explore the cost difference in a real app. Also seeing how many times in a modern app this is actually invoked. Many of the name/toString related work are from another era, I suspect the landscape has changed

**Comment by krisselden** [comment]
Majority of things that don't go through the container report as unknown because most stuff is now in modules not on a namespace.

**Comment by locks** [comment]
Can we get an update on this? cc @stefanpenner @krisselden

**Comment by rwjblue** [comment]
Fixed in 2.5.1
